### PR TITLE
Project visibility

### PR DESCRIPTION
**Pull request by rgachuhi**
This PR adds a feature to allow course publishers the ability to control who is allowed to create course instances for a particular course project. By default, any author to the course project has the ability to login to their Torus linked LMS, select a course project and build a course instance. This feature allow additional controls, whereby a publisher can choose to give global builder access or limit the access to only a select group of users or institutions.

Requires mix deps.get (updates ecto_sql to v3.5.2)

### CODE REVIEW

**Review by darrensiegel — CHANGES_REQUESTED** [review]
A couple of small code changes that I pointed out inline.

Also, I believe strongly that we need to move the Visibility LiveView *out* of the current page that it is located within.  Take a look at the attached screenshot of what this looks like as one starts making changes to a course.  The list of outstanding updates pushes the visibility view down, and will be difficult for users when that list becomes long.  I'm fine with the Visibility view being moved to the project Overview view, and placed under "Collaborators". 

Screen Shot 2021-02-05 at 9 17 40 AM

File: lib/oli/institutions.ex [hunk 1]
```diff
@@ -411,4 +411,14 @@ defmodule Oli.Institutions do
     end)
   end
 
+  @doc """
+  Searches for a list of Institution with an name matching a wildcard pattern
+  """
+  def search_authors_matching(query) do
```
[the remark this answers is not in the document]
Comment (rgachuhi) [reply lib/oli/institutions.ex]: Name changed

File: lib/oli_web/templates/project/insights.html.eex [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <div class="insights container">
   <div class="row">
     <div class="col-12">
-      <h2>Insights</h1>
+      <h1>Insights</h1>
```
[the remark this answers is not in the document]
Comment (rgachuhi) [reply lib/oli_web/templates/project/insights.html.eex:4]: 👍

File: priv/repo/migrations/20210128145315_create_project_visibilities.exs [hunk 3]
```diff
@@ -0,0 +1,22 @@
+defmodule Oli.Repo.Migrations.CreateProjectVisibilities do
+  use Ecto.Migration
+
+  def change do
+    create table(:project_visibilities) do
+      add :project_id, :integer
```
[the remark this answers is not in the document]
Comment (rgachuhi) [reply priv/repo/migrations/20210128145315_create_project_visibilities.exs]: switched fields to foreign key refs

**Comment by rgachuhi** [comment]
> A couple of small code changes that I pointed out inline.
> 
> Also, I believe strongly that we need to move the Visibility LiveView _out_ of the current page that it is located within. Take a look at the attached screenshot of what this looks like as one starts making changes to a course. The list of outstanding updates pushes the visibility view down, and will be difficult for users when that list becomes long. I'm fine with the Visibility view being moved to the project Overview view, and placed under "Collaborators".
> 
> Screen Shot 2021-02-05 at 9 17 40 AM

I have moved the package visibility UI into the project overview page.

**Review by darrensiegel — APPROVED** [review]

**Pull request merged**